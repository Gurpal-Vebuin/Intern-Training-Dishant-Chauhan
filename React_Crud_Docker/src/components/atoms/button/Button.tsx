import React from "react";
import { ButtonPropsType } from "../../../types/interfaces/interface";
import { RegisterButton } from "../../styles/navbar/NavbarStyles";

const Button: React.FC<ButtonPropsType> = ({
  type = "button",
  text,
  onClick,
  isDisabled = false,
  children,
}) => {
  return (
    <>
      <RegisterButton type={type} onClick={onClick} disabled={isDisabled}>
        {children || text}
      </RegisterButton>
    </>
  );
};

export default Button;
