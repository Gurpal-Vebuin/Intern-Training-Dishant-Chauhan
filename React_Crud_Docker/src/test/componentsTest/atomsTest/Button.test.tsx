import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import Button from "../../../components/atoms/button/Button";

describe("Button Component", () => {
  test("Renders Button with text", () => {
    render(<Button text="Click Me" />);
    const buttonElement = screen.getByText("Click Me");
    expect(buttonElement).toBeInTheDocument();
  });

  test("Handles click event", () => {
    const handleClick = jest.fn();
    render(<Button text="Submit" onClick={handleClick} />);

    const buttonElement = screen.getByText("Submit");
    fireEvent.click(buttonElement);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test("Renders Button as disabled", () => {
    render(<Button text="Disabled" isDisabled />);

    const buttonElement = screen.getByText("Disabled");
    expect(buttonElement).toBeDisabled();
  });

  test("Renders children instead of text", () => {
    render(
      <Button>
        <span>Custom Child</span>
      </Button>
    );

    const childElement = screen.getByText("Custom Child");
    expect(childElement).toBeInTheDocument();
  });

  test("Generating a Snapshot", () => {
    const { asFragment } = render(
      <Button>
        <span>Custom Child</span>
      </Button>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
