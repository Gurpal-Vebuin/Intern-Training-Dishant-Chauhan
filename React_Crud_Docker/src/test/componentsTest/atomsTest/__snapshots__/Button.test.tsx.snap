// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component Generating a Snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-dycHyt loRgPy"
    type="button"
  >
    <span>
      Custom Child
    </span>
  </button>
</DocumentFragment>
`;
